### DISCUSSION TITLE
Disconnect Events

### DISCUSSION THREAD

**Discussion by BButner** [hyperium/tonic#1243] [Q&A]
Is there any built-in functionality for me to listen for an event when the client disconnects?

From the server-side, I don't care if a client disconnects. But on the client side, I would like to be able to respond to disconnections, attempt to reconnect, and re-launch the stream that I have going during the lifetime of the client.

(I'm still somewhat new to Rust, so I'm sorry if I've missed anything obvious!)

**Comment by LucioFranco** [comment]
No, the only way is to setup a streaming request and when that stream closes you know you disconnected.
